endaq/calc/shock.py

```diff
@@ -17,29 +17,186 @@
 from endaq.calc import utils
 
 
-def _rel_displ_transfer_func(
-    omega: float, damp: float = 0.0, dt: float = 1.0
-) -> scipy.signal.ltisys.TransferFunctionDiscrete:
+def _abs_accel_coeffs(omega, Q, T):
     """
-    Generate the transfer function
-        H(s) = L{z(t)}(s) / L{y"(t)}(s) = (1/s²)(Z(s)/Y(s))
-    for the PDE
+    Calculate the coefficients of the Z-domain transfer function for the
+    absolute acceleration according to ISO 18431-4
+
+    :param omega: the natural frequency of the system
+    :param Q: the quality factor of the system
+    :param T: the time step in seconds
+    :return: the coefficients of the Z-domain transfer function b, a
+
+    .. seealso::
+
+        - `ISO 18431-4 Mechanical vibration and shock — Signal processing — Part 4: Shock-response spectrum analysis`
+          Explicit implementations of digital filter coefficients for shock spectra.
+    """
+    A = omega*T/(2.*Q)
+    B = omega*T*np.sqrt(1. - 1./(4.*(Q**2)))
+
+    b = (
+        1. - np.exp(-A)*np.sin(B)/B,
+        2.*np.exp(-A)*(np.sin(B)/B - np.cos(B)),
+        np.exp(-2*A) - np.exp(-A)*np.sin(B)/B,
+        )
+    a = (
+        1.,
+        -2.*np.exp(-A)*np.cos(B),
+        np.exp(-2.*A),
+        )
+
+    return b, a
+
+
+def abs_accel(accel: pd.DataFrame, omega: float, damp: float = 0.0) -> pd.DataFrame:
```

Comment:
abbreviated form: as-is 👍
non-abbreviated form:
```suggestion
def absolute_acceleration(accel: pd.DataFrame, omega: float, damp: float = 0.0) -> pd.DataFrame:
```